lib/client.js

```diff
@@ -438,14 +438,27 @@
 
     socket.on('error', function(e) {
 
+      var socketError = e;
+
+      socketError.code = e.error.code;
```

Comment:
I've been thinking about this - I think it would be better to assign e.error to socketError so that we don't leak references to websockets etc., which should not be of external interest.